tests/head_response_empty_body.rs

```diff
@@ -0,0 +1,23 @@
+#![feature(futures_api, async_await)]
+
+use futures::executor::block_on;
+use http_service::Body;
+use http_service_mock::make_server;
+use tide::{error::ResultExt, Context};
+
+async fn ok(_cx: Context<()>) -> String {
+    String::from("this shouldn't exist in the body of a HEAD response")
```

Comment:
I think if we want to make this a persistent test, just having this panic might be reasonable to signify it should never be executed.

Reply:
We could do that... Maybe something like this, I'm guessing?
```suggestion
    panic!("should not go here");
    String::from("this shouldn't exist in the body of a HEAD response")
```